test/e2e/azure_logcollector.go

```diff
@@ -177,61 +176,6 @@ func getHostname(m *clusterv1.Machine, isWindows bool) string {
 	return hostname
 }
 
-func getAzureCluster(ctx context.Context, managementClusterClient client.Client, namespace, name string) (*infrav1.AzureCluster, error) {
```

Comment:
I was able to re-use one of these existing funcs, but needed another one that wasn't there, so took the opportunity to add that (`getAzureManagedCluster`) in the more general-purpose `aks.go` source file rather than here. And because of the re-use of `getAzureManagedControlPlane` it seems like we can move all of these (reusable) funcs to `aks.go` from this log collection-specific source file.